client/src/main/scala/client/BackPressuredWebSocketClient.scala

```diff
@@ -1,58 +1,67 @@
+import monix.execution.{Ack, Cancelable, Scheduler}
+import monix.reactive.observers.Subscriber
+import monix.reactive.{Observable, Observer}
 import org.reactivestreams.Subscription
 import org.scalajs.dom.raw.MessageEvent
 import org.scalajs.dom.{CloseEvent, ErrorEvent, Event, WebSocket}
+
 import scala.concurrent._
-import concurrent.duration._
+import scala.concurrent.duration._
 
 
-final class BackPressuredWebSocketClient private (url: String)
-  extends Observable[String] { self =>
+final class BackPressuredWebSocketClient private(url: String)
+  extends Observable[String] {
+  self =>
 
   private def createChannel(webSocket: WebSocket) =
-    Observable.create[String] { subscriber =>
-      import subscriber.scheduler
-      val downstream = Observer.toReactiveSubscriber(subscriber)
-
-      try {
-        webSocket.onopen = (event: Event) => {
-          downstream.onSubscribe(new Subscription {
-            def cancel(): Unit =
-              closeConnection(webSocket)
-
-            def request(n: Long): Unit = {
-              webSocket.send(n.toString)
-            }
-          })
-        }
-
-        webSocket.onerror = (event: ErrorEvent) => {
-          downstream.onError(BackPressuredWebSocketClient.Exception(event.message))
-        }
-
-        webSocket.onclose = (event: CloseEvent) => {
-          downstream.onComplete()
+    new Observable[String] {
+      override def unsafeSubscribeFn(subscriber: Subscriber[String]): Cancelable = {
+        import subscriber.scheduler
+        val downstream = Observer.toReactiveSubscriber(subscriber)
+
+        try {
+          webSocket.onopen = (event: Event) => {
+            downstream.onSubscribe(new Subscription {
+              def cancel(): Unit =
+                closeConnection(webSocket)
+
+              def request(n: Long): Unit = {
+                webSocket.send(n.toString)
+              }
+            })
+          }
+
+          webSocket.onerror = (event: ErrorEvent) => {
+            downstream.onError(BackPressuredWebSocketClient.Exception(event.message))
+          }
+
+          webSocket.onclose = (event: CloseEvent) => {
+            downstream.onComplete()
+          }
+
+          webSocket.onmessage = (event: MessageEvent) => {
+            downstream.onNext(event.data.asInstanceOf[String])
+          }
+        } catch {
+          case ex: Throwable =>
+            downstream.onError(ex)
         }
 
-        webSocket.onmessage = (event: MessageEvent) => {
-          downstream.onNext(event.data.asInstanceOf[String])
-        }
-      }
-      catch {
-        case ex: Throwable =>
-          downstream.onError(ex)
+        BooleanCancelable()
       }
     }
 
   private def closeConnection(webSocket: WebSocket)(implicit s: Scheduler): Unit = {
     if (webSocket != null && webSocket.readyState <= 1)
-      try webSocket.close() catch { case _: Throwable => () }
+      try webSocket.close() catch {
+        case _: Throwable => ()
```

Comment:
Here we shouldn't catch `Throwable`. Not sure why I was doing it. Maybe it's because this runs on top of Javascript therefore `NonFatal` isn't so useful anymore. But we shouldn't do this anyway. And errors should be logged.

In other words `catch NonFatal(ex) => s.reportError(ex)` because otherwise we are teaching people bad practices.